src/Visitor/GraphNavigator.php

```diff
@@ -54,6 +56,12 @@ public function visit($value): void
 
     private function visitObject($value): void
     {
+        if ($value instanceof AnonymizableInterface) {
```

Comment:
Wouldn't it be better to have this as an extension/event subscriber?

Reply:
As described I don't see any other options to guarantee that the functions will be executed as first and last one if it's part of the event system.  Either `stopPropagation` or the order of priorities would prevent that.
Likewise, there is no other place where every object is checked